services/common/src/components/help/HelpGuide.tsx

```diff
@@ -0,0 +1,202 @@
+  const [isEditMode, setIsEditMode] = useState(false);
+  const canEditHelp = useSelector((state) => userHasRole(state, USER_ROLES.role_edit_helpdesk));
+  const helpGuide = useSelector(getHelpByKey(helpKey, pageTab)) ?? {};
+  const { help_guid, help_key } = helpGuide;
+  const hasHelpGuide = Boolean(help_guid);
+  const defaultGuide = help_key === EMPTY_HELP_KEY;
+  const { content = "" } = helpGuide ?? {};
+  const [isLoaded, setIsLoaded] = useState(hasHelpGuide);
+  const unformattedTitle = pageTab ?? helpKey;
+  const title = formatSnakeCaseToSentenceCase(unformattedTitle);
+
+  const showDrawer = () => setOpen(true);
+  const hideDrawer = () => {
+    setOpen(false);
+    setIsEditMode(false);
+  };
+
+  const handleFetchData = () => {
+    dispatch(fetchHelp({ helpKey, system })).then(() => setIsLoaded(true));
+  };
+
+  useEffect(() => {
+    dispatch(clearHelp(helpKey));
+    handleFetchData();
+  }, [helpKey]);
+
+  const handleDeleteGuide = () => {
+    deleteConfirmWrapper("Help Guide", () =>
+      dispatch(deleteHelp({ helpKey, help_guid })).then(() => {
+        setIsEditMode(false);
+      })
+    );
+  };
+
+  const triggerSubmit = () => {
+    dispatch(submit(FORM.EDIT_HELP_GUIDE));
+  };
+
+  const handleSaveGuide = (values) => {
+    const page_tab = values.page_tab ?? HELP_GUIDE_ALL_TABS;
+    const data = { ...values, page_tab, system, help_key: helpKey };
+
+    const payload = { helpKey, data };
+    const forNewTab =
+      helpGuide.page_tab === HELP_GUIDE_ALL_TABS && values.page_tab !== HELP_GUIDE_ALL_TABS;
+    const saveFunction =
+      defaultGuide || forNewTab
+        ? () => dispatch(createHelp(payload))
+        : () => dispatch(updateHelp(payload));
+
+    saveFunction().then((resp) => {
+      if (resp.payload) {
+        setIsEditMode(false);
+      }
+    });
+  };
+
+  const getFooterButtons = () => {
+    const editButton = (
+      <Button type="primary" onClick={() => setIsEditMode(true)}>
+        {hasHelpGuide ? "Edit Help Guide" : "Create Help Guide"}
+      </Button>
+    );
+    const submitButton = (
+      <Button type="primary" onClick={triggerSubmit}>
+        Publish Help Guide
+      </Button>
+    );
+    const deleteButton = (
+      <Button disabled={!hasHelpGuide || defaultGuide} onClick={handleDeleteGuide}>
+        Delete this Guide
+      </Button>
+    );
+    const buttons = isEditMode ? (
+      <>
+        <Col>{deleteButton}</Col>
+        <Col>{submitButton}</Col>
+      </>
+    ) : (
+      <Col>{editButton}</Col>
+    );
+    const justify = isEditMode ? "space-between" : "end";
+    return <Row justify={justify}>{buttons}</Row>;
+  };
+  const initialValues =
+    hasHelpGuide && !defaultGuide ? helpGuide : { help_key: helpKey, page_tab: pageTab };
+  const mainContent = isEditMode ? (
+    <HelpGuideForm
+      initialValues={initialValues}
+      handleSaveGuide={handleSaveGuide}
+      pageTab={pageTab}
+    />
+  ) : (
+    parse(content)
```

Comment:
I'm wondering if this might be vulnerable for an XSS injection? Seeing as it just renders the html stored in the DB directly.
If so, we might want to sanitize it using something like `sanitize-html` before we render it

Reply:
`DomPurify` also looks good!

Reply:
Just added!
Added a test on MS where the rendered (dataMocks) content contains both a script tag and a click handler to verify that they're stripped out.